tmol/score/backbone_torsion/potentials/potentials.hh

```diff
@@ -0,0 +1,189 @@
+
+  dV_dpsiatm.row(0) = dVdphipsi[1] * psiang.dV_dI / bbstep[1];
+  dV_dpsiatm.row(1) = dVdphipsi[1] * psiang.dV_dJ / bbstep[1];
+  dV_dpsiatm.row(2) = dVdphipsi[1] * psiang.dV_dK / bbstep[1];
+  dV_dpsiatm.row(3) = dVdphipsi[1] * psiang.dV_dL / bbstep[1];
+  return {V, dV_dphiatm, dV_dpsiatm};
+}
+
+template <tmol::Device D, typename Real, typename Int>
+def omega_V_dV(CoordQuad omega, Real K)->tuple<Real, CoordQuad> {
+  Real V;
+  Real dVdomega;
+  CoordQuad dV_domegaatm;
+
+  auto omegaang = dihedral_angle<Real>::V_dV(
+      omega.row(0), omega.row(1), omega.row(2), omega.row(3));
+
+  // note: the angle returned by dihedral_angle is in [-pi,pi]
+  Real omega_offset = omegaang.V;
+  if (omega_offset > 0.5 * EIGEN_PI) {
+    omega_offset -= EIGEN_PI;
+  } else if (omega_offset < -0.5 * EIGEN_PI) {
+    omega_offset += EIGEN_PI;
+  }
+
+  V = K * omega_offset * omega_offset;
+  dVdomega = 2 * K * omega_offset;
+
+  dV_domegaatm.row(0) = omegaang.dV_dI;
+  dV_domegaatm.row(1) = omegaang.dV_dJ;
+  dV_domegaatm.row(2) = omegaang.dV_dK;
+  dV_domegaatm.row(3) = omegaang.dV_dL;
+
+  return {V, dVdomega * dV_domegaatm};
+}
+
+// bb-dep version of omega
+template <tmol::Device D, typename Real, typename Int>
+def omega_bbdep_V_dV(
+    CoordQuad phi,
+    CoordQuad psi,
+    CoordQuad omega,
+    TensorAccessor<Real, 2, D> coeffs_mu,
+    TensorAccessor<Real, 2, D> coeffs_sig,
+    Real2 bbstart,
+    Real2 bbstep,
+    Real K)
+    ->tuple<Real, CoordQuad, CoordQuad, CoordQuad> {
+  const Real pi = EIGEN_PI;
+  Real V;
+  Real dVdomega, dVdphi = 0.0, dVdpsi = 0.0;
+  CoordQuad dV_domegaatm, dV_dphiatm, dV_dpsiatm;
+
+  auto omegaang = dihedral_angle<Real>::V_dV(
+      omega.row(0), omega.row(1), omega.row(2), omega.row(3));
+
+  // note: the angle returned by dihedral_angle is in [-pi,pi]
+  if (omegaang.V > -0.5 * EIGEN_PI && omegaang.V < 0.5 * EIGEN_PI) {
+    // cis omega, use simple version
+    tie(V, dV_domegaatm) = omega_V_dV<D, Real, Int>(omega, K);
+    dV_dphiatm.setZero();
+    dV_dpsiatm.setZero();
+  } else {
+    auto phiang = dihedral_angle<Real>::V_dV(
+        phi.row(0), phi.row(1), phi.row(2), phi.row(3));
+    auto psiang = dihedral_angle<Real>::V_dV(
+        psi.row(0), psi.row(1), psi.row(2), psi.row(3));
+
+    Real mu, sig;
+    Real2 dmudphipsi, dsigdphipsi;
+
+    Real2 phipsi_idx;
+    phipsi_idx[0] = (phiang.V - bbstart[0]) / bbstep[0];
+    phipsi_idx[1] = (psiang.V - bbstart[1]) / bbstep[1];
+
+    tie(mu, dmudphipsi) =
+        tmol::numeric::bspline::ndspline<2, 3, D, Real, Int>::interpolate(
+            coeffs_mu, phipsi_idx);
+    tie(sig, dsigdphipsi) =
+        tmol::numeric::bspline::ndspline<2, 3, D, Real, Int>::interpolate(
+            coeffs_sig, phipsi_idx);
+
+    // energy
+    Real baseline = std::log(1. / (6. * sqrt(2. * pi)))
+                    - std::log(1. / (sig * sqrt(2. * pi)));
+    Real offset = omegaang.V * 180.0 / pi - mu;
+    if (offset < -180.0) {
+      offset += 360.0;
```

Comment:
oh weird! So, let's say that for a given phi/psi, the mu is 170. Then when omega is 175 it gets pushed toward 170. That makes sense to me. BUT if omega is cis and -5, then it's going to get pushed towards -10?? Is there really symmetry between mu for cis and trans omegas??

Reply:
when omega is cis it falls into the loop at 121 - 123 (no bb-dependence), 125-179 is run on trans only.

Reply:
I'll add comments clarifying this.